src/query_parser/parser.y

```diff
@@ -787,26 +779,28 @@ geo_filter(A) ::= LSQB param_any(B) param_any(C) param_any(D) param_any(E) RSQB.
 query ::= expr(A) ARROW LSQB vector_query(B) RSQB . { // main parse, hybrid query as entire query case.
   setup_trace(ctx);
   switch (B->vn.vq->type) {
-    case VECSIM_QT_TOPK:
-      B->vn.vq->topk.order = BY_SCORE;
+    case VECSIM_QT_KNN:
+      B->vn.vq->knn.order = BY_SCORE;
       break;
   }
   ctx->root = B;
-  QueryNode_AddChild(B, A);
+  if (A) {
+    QueryNode_AddChild(B, A);
+  }
 }
 
 query ::= star ARROW LSQB vector_query(B) RSQB . { // main parse, simple vecsim search as entire query case.
   setup_trace(ctx);
   switch (B->vn.vq->type) {
-    case VECSIM_QT_TOPK:
-      B->vn.vq->topk.order = BY_SCORE;
+    case VECSIM_QT_KNN:
+      B->vn.vq->knn.order = BY_SCORE;
       break;
   }
   ctx->root = B;
 }
 
 // Vector query opt. 1 - full query.
-vector_query(A) ::= vector_command(B) vector_attribute_list(C) AS param_term(D). {
+vector_query(A) ::= vector_command(B) vector_attribute_list(C) vector_score_filed(D). {
```

Comment:
In various locations
```suggestion
vector_query(A) ::= vector_command(B) vector_attribute_list(C) vector_score_field(D). {
```

Reply:
I accidentally fixed that in the other PR. I’ll leave this typo here for now